### ISSUE TITLE
@a[1, ...] gives incorrect results

### ISSUE THREAD

**Issue by zoffixznet** [rakudo/rakudo#2185]
This appears to have happened after merge of https://github.com/rakudo/rakudo/pull/1844 (ping @MasterDuke17 )

First, specifying a list in the index now treats that list as Cool instead of as a list of indices:

```
<Zoffix_> c: 2018.03,HEAD my @a = ^100; dd @a[1,(1, 2, 3)]
<committable6> Zoffix_, ¦2018.03: «(1, (1, 2, 3))␤» ¦HEAD(1978a2f): «(1, 3)␤»
```

The same problem existed with a Range, but the behaviour changed in https://github.com/rakudo/rakudo/commit/1978a2fc1d33aea03547ea47130102ec791d436f . IMO it's wrong to use the static optimizer to fix bugs, as then the bugs still exist when the optimizer is turned off. Also, the fix doesn't fix the issue correctly, because the produced result is a flat list, as if the Range was slipped, not a nested one:

```
<Zoffix_> c: 2018.03,HEAD my @a = ^100; dd @a[1, 1..3]
<committable6> Zoffix_, ¦2018.03: «(1, (1, 2, 3))␤» ¦HEAD(1978a2f): «(1, 1, 2, 3)␤»
```

**Comment by lizmat** [comment]
Fixed with 1978a2fc1d33aea03547ea

**Comment by lizmat** [comment]
Is should read what the issue is  :-)

**Comment by lizmat** [comment]
Fixed with a3a8712560b44820846a
